### PR TITLE
Reschedule functionality and Reschedule Test added

### PR DESCRIPTION
**Pull request by agentic-k**
## Description
Reschedule functionality added to the scheduler.re

## Related to
Scheduler.re

## Checklist
- [x] added unit tests
- [x] run the examples

### CODE REVIEW

File: src/Scheduler.rei [hunk 1]
```diff
@@ -17,4 +17,5 @@ type t;
 
 let create: unit => t;
 let add: (t, job) => jobId;
-let remove: (t, jobId) => unit;
\ No newline at end of file
+let remove: (t, jobId) => unit;
+let reschedule: (t, job, jobId) => jobId;
```
Comment (alireza-a) [inline src/Scheduler.rei:21]: the user should only pass in a new recurrence

**Review by alireza-a — CHANGES_REQUESTED** [review]

File: src/Scheduler.rei (hunk 1 above)
Comment (selfeki) [reply src/Scheduler.rei:21]: You mean replace the `job` param with `recurrence`?

Comment (alireza-a) [reply src/Scheduler.rei:21]: I shouldn't need to know what the invokation is to reschedule job

Comment (selfeki) [reply src/Scheduler.rei:21]: So Kunal, please pass in `recurrence` instead of `job` and make the related changes inside the method itself.

**Pull request closed without merging**